File: 14-TypeScript/Grey Multi-Tenant/packages/grey-r/tests/testthat/test-options.R
test_that("GreyOptions$new creates options correctly", {
  opts <- GreyOptions$new(host = "api.example.com")
  
  expect_equal(opts$host, "api.example.com")
  expect_equal(opts$port, 443)
  expect_true(opts$use_tls)
  expect_equal(opts$timeout, 30)
  expect_equal(opts$headers, list())
})

test_that("GreyOptions$local creates local options", {
  opts <- GreyOptions$local(3000)
  
  expect_equal(opts$host, "localhost")
  expect_equal(opts$port, 3000)
  expect_false(opts$use_tls)
})

test_that("GreyOptions$production creates production options", {
  opts <- GreyOptions$production("api.grey.com")
  
  expect_equal(opts$host, "api.grey.com")
  expect_equal(opts$port, 443)
  expect_true(opts$use_tls)
})

test_that("base_url returns correct URL", {
  local <- GreyOptions$local(8080)
  prod <- GreyOptions$production("api.grey.com")
  
  expect_equal(local$base_url(), "http://localhost:8080")
  expect_equal(prod$base_url(), "https://api.grey.com:443")
})

test_that("with_timeout creates new options", {
  original <- GreyOptions$local()
  modified <- original$with_timeout(120)
  
  expect_equal(original$timeout, 30)
  expect_equal(modified$timeout, 120)
  expect_equal(modified$host, original$host)
})

test_that("with_headers merges headers", {
  original <- GreyOptions$new(host = "api.grey.com", headers = list("X-A" = "a"))
  modified <- original$with_headers(list("X-B" = "b"))
  
  expect_equal(length(original$headers), 1)
  expect_equal(length(modified$headers), 2)
  expect_equal(modified$headers$`X-A`, "a")
  expect_equal(modified$headers$`X-B`, "b")
})

File: 14-TypeScript/Grey Multi-Tenant/packages/grey-julia/test/test_options.jl
using Test
using GreySDK.Options

@testset "Options" begin
    @testset "GreyOptions construction" begin
        opts = GreyOptions(host="api.grey.com")
        @test opts.host == "api.grey.com"
        @test opts.port == 443
        @test opts.use_tls == true
        @test opts.timeout == 30
        @test isempty(opts.headers)
        
        opts_custom = GreyOptions(
            host="localhost",
            port=8080,
            use_tls=false,
            timeout=60,
            headers=Dict("X-Custom" => "value")
        )
        @test opts_custom.host == "localhost"
        @test opts_custom.port == 8080
        @test opts_custom.use_tls == false
        @test opts_custom.timeout == 60
        @test opts_custom.headers["X-Custom"] == "value"
    end
    
    @testset "local_options" begin
        opts = local_options()
        @test opts.host == "localhost"
        @test opts.port == 8080
        @test opts.use_tls == false
        
        opts_custom_port = local_options(3000)
        @test opts_custom_port.port == 3000
    end
    
    @testset "production_options" begin
        opts = production_options("api.grey.com")
        @test opts.host == "api.grey.com"
        @test opts.port == 443
        @test opts.use_tls == true
        
        opts_custom_port = production_options("api.grey.com", 8443)
        @test opts_custom_port.port == 8443
    end
    
    @testset "base_url" begin
        local_opts = local_options(8080)
        @test base_url(local_opts) == "http://localhost:8080"
        
        prod_opts = production_options("api.grey.com")
        @test base_url(prod_opts) == "https://api.grey.com:443"
    end
    
    @testset "with_timeout" begin
        opts = local_options()
        new_opts = with_timeout(opts, 120)
        @test new_opts.timeout == 120
        @test opts.timeout == 30  # Original unchanged
    end
    
    @testset "with_headers" begin
        opts = local_options()
        new_opts = with_headers(opts, Dict("Authorization" => "Bearer token"))
        @test new_opts.headers["Authorization"] == "Bearer token"
        @test isempty(opts.headers)  # Original unchanged
    end
end
